tests/integration/targets/archive/tasks/main.yml

```diff
@@ -376,6 +378,80 @@
       - exclusion_patterns_result is changed
       - "'bar.txt' not in exclusion_patterns_result.archived"
 
+- name: Test that excluded paths do not influence archive root
+  archive:
+    path:
+      - "{{ output_dir }}/sub/subfile.txt"
+      - "{{ output_dir }}"
+    exclude_path:
+      - "{{ output_dir }}"
+    dest: "{{ output_dir }}/test-archive-root.tgz"
+  register: archive_root_result
+
+- name: Assert that excluded paths do not influence archive root
+  assert:
+    that:
+      - archive_root_result.arcroot != output_dir
+
+- name: Remove archive root test
+  file:
+    path: "{{ output_dir }}/test-archive-root.tgz"
+    state: absent
+
+- name: Test Single Target with format={{ item }}
+  archive:
+    path: "{{ output_dir }}/foo.txt"
+    dest: "{{ output_dir }}/test-single-gzip.{{ item }}"
+    format: "{{ item }}"
+  register: "single_target_test"
+  loop:
+    - zip
+    - tar
+    - gz
+    - bz2
+    - xz
+
+- name: Remove single target test with format={{ item }}
```

Comment:
I'd recommend having this task *after* the `assert` below. If someone breaks something one year from now, the failing assertion will leave the intermediate files behind, helping with the debugging process ;-)